rest_framework/permissions.py

```diff
@@ -107,13 +107,19 @@ def get_required_permissions(self, method, model_cls):
         return [perm % kwargs for perm in self.perms_map[method]]
 
     def has_permission(self, request, view):
-        try:
-            queryset = view.get_queryset()
-        except AttributeError:
-            queryset = getattr(view, 'queryset', None)
-        except AssertionError:
-            # view.get_queryset() didn't find .queryset
-            queryset = None
+        getter = getattr(view, 'get_queryset', None)
+        if getter is not None:
+            try:
+                queryset = view.get_queryset()
+            except AssertionError:
+                # The default implementation of view.get_queryset()
+                # didn't find the view.queryset attribute.
+                queryset = None
+        else:
+            try:
+                queryset = view.queryset
+            except AttributeError:
+                queryset = None
```

Comment:
> but its implementation might raise AttributeError during execution.

That's such a crazy edge case that I don't really care about it. I'd prefer clarity of this code, and we'd still be using sensible enough behavior in that case. (Plus we'd expect the error to be raised elsewhere such as when initially using the queryset in the view)